Add VPC Cloudformation template base.

ablundgren opened the pull request:
This is a cloudformation template and readme that can be used to launch a cluster within a VPC.  It only works in a single region.  It supports multiple availability zones.  You must edit the template and supply the required VPC networking configuration.

ablundgren committed:
Added Cloudformation template base

ablundgren committed:
Update README.md

ablundgren committed:
Update README.md

joaquincasares commented on community_cloud/README.md:
```diff
@@ -0,0 +1,36 @@
+- If you need machine types other than the m1.xlarge, you will need to add the configuration for the block mapping.
+- Your cluster (VPC) must be able to interact with sites on the internet to bootstrap the cluster.
+- The template does not prompt for all supported parameters. 
+- Deploying the DSE will require modifying the template to support the username and password parameters.
```
Could you mention that at present DSE 4.0 still does not default to vnodes? And until it does, the AMI will not automatically use vnodes.

joaquincasares commented on community_cloud/cassandra-vpc.template:
```diff
@@ -0,0 +1,110 @@
+        "Type" : "String",
+        "Default" : "m1.xlarge",
+        "AllowedValues" : ["m1.small", "m1.large", "m1.xlarge"],
+        "Description" : "Enter m1.small, m1.large or m1.xlarge. Default is <m1.xlarge>"
```
The AMI will not actually allow m1.smalls to be provisioned. Could you remove that instance type?

joaquincasares commented on community_cloud/cassandra-vpc.template:
```diff
@@ -0,0 +1,110 @@
+        "Type" : "String",
+        "Default" : "community",
+        "AllowedValues" : ["community", "enterprise"],
+        "Description" : "community or enterprise. Default is <community>"
```
I would feel happier if we commented out the enterprise option for now, if not we're allowing people to potentially setup a badly configured cluster, which is against the purpose of the AMI.

joaquincasares commented on community_cloud/cassandra-vpc.template:
```diff
@@ -0,0 +1,110 @@
+      },
+      "release" : {
+        "Type" : "String",
+        "Default" : "2.0.3",
```
Newest version is already 2.0.5. Is there a way to leave this blank and not generate a --release flag? If not, can you update the release number?

joaquincasares commented on community_cloud/cassandra-vpc.template:
```diff
@@ -0,0 +1,110 @@
+          }
+        },
+        "InstanceType" : "m1.xlarge",
+        "BlockDeviceMappings" : [
```
I'm not sure this is required since the AMI's should already have the blocking baked in. https://github.com/riptano/ComboAMI/blob/2.5/presetup/setup.md#baking-script-ebs-backed

If it is, could you add all the way to sdy: ephemeral23? It's my understanding that you can overdo the block device mappings, but it won't matter unless you have those ephemeral devices. A simple test of a m1.xlarge with the extended block mappings should suffice.

If that works, then you can edit the caveat above regarding the block device mapping.

joaquincasares commented:
Could you make the requested edit's then I'll give it another pass? Thanks again for the script!

ablundgren committed:
Update README.md

DSE 4

ablundgren committed:
Update cassandra-vpc.template

Removed invalid machine type.
Removed default release version, you must specify the version you want now.
Removed ability to specify enterprise. Added comment to note that you must work on the template to enable it.

ablundgren committed:
Drive mapping not required.

As noted in one of the pull request comments, the drive mapping was not required.

ablundgren committed:
Update README.md

ablundgren commented:
The edits are done.  (If you can't see all of them let me know and I will see if I can track down why.  I did them in place here on github.  I haven't done that before.  The last edits were today.)

ablundgren committed:
Should not have removed instance type with the drive mapping.

Also made the template correctly use the specified instance type parameter rather than hard coding it.

ablundgren committed:
fixed JSON syntax errors around template portions that must be replaced anyway.

joaquincasares commented:
This has now been committed and placed here, with a few minor changes.

Thanks again and sorry about the delay!

https://github.com/riptano/ComboAMI/tree/2.5/community_cloud

joaquincasares commented:
Also, I've deprecated the 2.5 AMI and released a new set of 2.5.1 AMIs.

You can find the json here: https://github.com/riptano/ComboAMI/blob/2.5/ami_ids.json, in case your scripts start breaking.
